File: python/fizz_buzz/main_test.py
from main import fizz_buzz


def test_fizz_buzz():
    for i in range(0, 100):
        result = fizz_buzz(i)

        if i % 3 == 0 and i % 5 == 0:
            assert(result == "FizzBuzz")
        elif i % 3 == 0:
            assert(result == "Fizz")
        elif i % 5 == 0:
            assert(result == "Buzz")
        else:
            assert(result == str(i))


if __name__ == '__main__':
    test_fizz_buzz()

File: go/fizzbuzz/main_test.go
package main

import (
	"strconv"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestFizzBuzz(t *testing.T) {
	for i := 0; i < 100; i++ {
		result := FizzBuzz(i)

		if i%3 == 0 && i%5 == 0 {
			assert.Equal(t, "FizzBuzz", result)
		} else if i%3 == 0 {
			assert.Equal(t, "Fizz", result)
		} else if i%5 == 0 {
			assert.Equal(t, "Buzz", result)
		} else {
			assert.Equal(t, strconv.Itoa(i), result)
		}
	}
}
